v3/src/components/graph/adornments/movable-value/movable-value.tsx

```diff
@@ -1,116 +1,270 @@
-      model.setValue(xScale.invert(event.x))
-    }, [model, xScale])
+      // Clear any previously added elements
+      valueObjects.current = []
+      const selection = select(valueRef.current)
+      selection.html(null)
+      select(`#${containerId}`).selectAll("div").remove()
 
-  // Add the behavior to the line cover
-  useEffect(function addBehaviors() {
-    valueObject.cover?.call(drag().on('drag', dragValue))
-  }, [valueObject, dragValue])
+      for (let i = 0; i < valueSet.length; i++) {
+        const newValueObject: any = {}
+        const { x1, x2, y1, y2 } = determineLineCoords(valueSet[i])
+        const orientationClass = isVertical.current ? "vertical" : "horizontal"
 
-  // Make the line and its cover segments just once
-  useEffect(function createElements() {
-    const selection = select(valueRef.current),
-      newValueObject: any = {}
-    newValueObject.line = selection.append('line')
-      .attr('class', 'movable-value')
-      .attr('transform', transform)
-    newValueObject.cover = selection.append('line')
-      .attr('class', 'movable-value-cover')
-      .attr('transform', transform)
-    newValueObject.valueLabel = select(kGraphClassSelector).append('div')
-      .attr('class', 'movable-value-label')
-    setValueObject(newValueObject)
+        newValueObject.rect = selection.append("rect")
+          .attr("class", `movable-value-rect ${orientationClass}`)
+          .attr("transform", transform)
+          .attr("x", isVertical.current ? x1 - 3 : x1)
+          .attr("y", isVertical.current ? y1 : y1 - 3)
+        newValueObject.line = selection.append("line")
+          .attr("class", `movable-value ${orientationClass}`)
+          .attr("transform", transform)
+          .attr("x1", x1)
+          .attr("y1", y1)
+          .attr("x2", x2)
+          .attr("y2", y2)
+        newValueObject.cover = selection.append("line")
+          .attr("class", `movable-value-cover ${orientationClass}`)
+          .attr("transform", transform)
+          .attr("x1", x1)
+          .attr("y1", y1)
+          .attr("x2", x2)
+          .attr("y2", y2)
+        newValueObject.valueLabel = select(`#${containerId}`).append("div")
+          .attr("class", `movable-value-label ${orientationClass}`)
+          .style("left", `${x1}px`)
+          .style("top", `${y1}px`)
+          .html(valueLabelString(valueSet[i]))
 
+        valueObjects.current = [...valueObjects.current, newValueObject]
+        addDragHandlers()
+        renderFills()
+      }
+    })
+  }, [addDragHandlers, renderFills, containerId, determineLineCoords, valueSet, transform, instanceKey])
+
+  useEffect(function cleanup() {
     return () => {
-      newValueObject.valueLabel
-        .transition()
-        .duration(1000)
-        .style('opacity', 0)
-        .end().then(() => {
-        newValueObject.valueLabel.remove()
-      })
+      valueObjects.current = []
     }
-  }, [transform])
+  }, [])
```

Comment:
This effect seems unnecessary since it's only going to run on unmount at which point the ref will be destroyed automatically.